### PR TITLE
Update specs based on new compilers and system default changes

### PR DESCRIPTION
**Pull request by rhornung67**
The title says it all

### CODE REVIEW

**Comment by rhornung67** [comment]
@adrienbernede This is still a WIP. Gathering info from app teams about compilers used in production builds. I will ping you when it's ready.

**Comment by rhornung67** [comment]
@adrienbernede @davidbeckingsale this should be ready to merge. It minimizes the shared configs to the system default and what production apps are using on each platform.

File: ruby-build-and-test.yml [hunk 1]
```diff
@@ -92,9 +92,9 @@ status_pending:
     - export pipeline_status="pending"
     - *status_report
 
-intel_19_1_2_gcc_8_5_0:
+intel_19_1_2_gcc_10_3_1:
   variables:
-    SPEC: "${PROJECT_RUBY_VARIANTS} %intel@19.1.2.gcc.8.5.0 ${PROJECT_RUBY_DEPS}"
+    SPEC: "${PROJECT_RUBY_VARIANTS} %intel@19.1.2.gcc.10.3.1 ${PROJECT_RUBY_DEPS}"
```
Comment (adrienbernede) [inline ruby-build-and-test.yml:97]: In Umpire Gcc @ 10.3.1 won't work with intel @ 19.1

Comment (adrienbernede) [reply ruby-build-and-test.yml:97]: I'll have to test this. I had quite clear results for my experiment using the system gcc (assuming it was 8.5.0) vs 10.3.1.

**Comment by adrienbernede** [comment]
Closing: merged in https://github.com/LLNL/radiuss-shared-ci/pull/29

**Pull request closed without merging**